command/operator_debug.go

```diff
@@ -692,18 +723,31 @@ func (c *OperatorDebugCommand) collectNomad(dir string, client *api.Client) erro
 	ns, _, err := client.Nodes().List(qo)
 	c.writeJSON(dir, "nodes.json", ns, err)
 
+	// CSI Plugins - /v1/plugins?type=csi
 	ps, _, err := client.CSIPlugins().List(qo)
 	c.writeJSON(dir, "plugins.json", ps, err)
 
-	vs, _, err := client.CSIVolumes().List(qo)
-	c.writeJSON(dir, "volumes.json", vs, err)
+	// Loop over each plugin - /v1/plugin/csi/:plugin_id
+	for _, p := range ps {
+		csiPlugin, _, _ := client.CSIPlugins().Info(p.ID, qo)
+		csiPluginFileName := fmt.Sprintf("csi-plugin-id-%s", p.ID)
```

Comment:
A plugin name isn't necessarily safe to use as a filepath. Ex. silly things like `aws/ebs` or `gcp-*-1` are acceptable names. We should coerce the ID to a "safe" path by swapping out "special characters" for `-` or `_`. (Pulling this out to a function we could reuse in the `helpers` package would be a nice-to-have too, but no need to do that in this PR.)

The same goes for volumes, below.

Reply:
I added 3 new helper functions for replacing invalid characters from a filename.  I'm using the bare minimum for Linux/Windows compliance, but I also added stricter variations which remove more punctuation or all non-ASCII characters.  Code is in writeBytes to ensure plugins, volumes, and all other potentially dynamic names are covered.

Reply:
looks good :+1:

Reply:
I also added a test to verify that plugin names such as those are in fact escaped.